File: backend/src/main/java/edu.cit.cararag.attendme/dto/requestdto/BulkAttendanceRequest.java
package edu.cit.cararag.attendme.dto.requestdto;

import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;
import java.util.Map;

@Data
public class BulkAttendanceRequest {
    
    @NotNull(message = "Class ID is required")
    private Long classId;
    
    @NotNull(message = "Date is required")
    private LocalDate date;
    
    @NotNull(message = "Attendance data is required")
    private Map<Long, String> attendanceData;
    
    private String remarks;
    
    @NotNull(message = "Marked by user ID is required")
    private Long markedById;
}

File: mobile/AttendMe/app/src/main/java/edu/cit/cararag/attendme/data/model/BulkAttendanceRequest.kt
package edu.cit.cararag.attendme.data.model

data class BulkAttendanceRequest(
    val classId: Long,
    val date: String,
    val attendanceList: List<AttendanceItem>
)

data class AttendanceItem(
    val studentId: Long,
    val status: String,
    val remarks: String? = null
)